rs_client/staging_client.py

```diff
@@ -0,0 +1,274 @@
+
+
+class StagingClient(RsClient):
+    """
+    Class to handle the staging process in rs-client-libraries
+
+    This class provides python methods to call the different endpoints of the rs-server-staging method
+
+    Remark: this class don't inherits from the owslib.ogcapi.processes.Processes class because the latter
+    doesn't provide wrapping for all endpoints defined in rs-server-staging (it only provides the  /processes
+    and /processes/{processId}/execution endpoints + it doesn't allow to manage apikey_header parameter which
+    is passed as an extra argument)
+    """
+
+    def __init__(self, rs_server_href: str | None, rs_server_api_key: str | None, owner_id: str | None, logger: Any):
+        """
+        Initialize the StagingClient parameters
+        """
+        super().__init__(rs_server_href, rs_server_api_key, owner_id, logger)
+        # Define logger
+        self.logger = logger
+        self.resource = "staging"
+        self.job_id: str | None = None
+
+    @property
+    def href_staging(self) -> str:
+        """
+        Return the RS-Server staging URL hostname.
+        This URL can be overwritten using the RSPY_HOST_STAGING env variable (used e.g. for local mode).
+        Otherwise it should just be the RS-Server URL.
+        """
+        if from_env := os.getenv("RSPY_HOST_STAGING", None):
+            return from_env.rstrip("/")
+        if not self.rs_server_href:
+            raise RuntimeError("RS-Server URL is undefined")
+        return self.rs_server_href.rstrip("/")
+
+    @property
+    def href_search(self) -> str:
+        """href for search"""
+        return ""
+
+    @property
+    def href_status(self) -> str:
+        """href for status"""
+        return ""
+
+    @property
+    def href_processes(self) -> str:
+        """
+        Url to get processes
+        """
+        return f"{self.href_staging}/processes"
+
+    @property
+    def href_resources(self) -> str:
+        """
+        Url for the staging process
+        """
+        return f"{self.href_staging}/processes/{self.resource}"
+
+    @property
+    def href_execute_process(self) -> str:
+        """
+        Url to execute a specific process
+        """
+        return f"{self.href_staging}/processes/{self.resource}/execution"
+
+    @property
+    def href_jobs(self) -> str:
+        """
+        Url to get running jobs
+        """
+        return f"{self.href_staging}/jobs"
+
+    @property
+    def href_job_status(self) -> str:
+        """
+        Url to get status of a specific job
+        """
+        return f"{self.href_staging}/jobs/{self.job_id}"
+
+    @property
+    def href_job_result(self) -> str:
+        """
+        Url to get the results of a specific job
+        """
+        return f"{self.href_staging}/jobs/{self.job_id}/results"
+
+    ############################
+    # Call RS-Server endpoints #
+    ############################
+
+    def run_staging(  # pylint: disable=too-many-locals
```

Comment:
@vprivat-ads can you confirm that all following endpoints:
GET /processes
GET /processes/{process}
POST /processes/{process}/execution
GET /jobs
GET /jobs/{job-id}
DELETE /jobs/{job-id}
GET /jobs/{job-id}/results

need to be wrapped in a rs-client function ?